samples/drivers/flash_shell/src/main.c

```diff
@@ -159,200 +160,210 @@ static int do_read(off_t offset, size_t len)
 		if (ret) {
 			goto err_read;
 		}
-		dump_buffer(buf, sizeof(buf));
+		dump_buffer(shell, buf, sizeof(buf));
 		len -= sizeof(buf);
 		offset += sizeof(buf);
 	}
 	ret = flash_read(flash_device, offset, buf, len);
 	if (ret) {
 		goto err_read;
 	}
-	dump_buffer(buf, len);
+	dump_buffer(shell, buf, len);
 	return 0;
 
  err_read:
-	printk("flash_read: %d\n", ret);
+	PR_SHELL(shell, "flash_read: %d\r\n", ret);
 	return 0;
 }
 
 /* Erase area, handling write protection and printing on error. */
-static int do_erase(off_t offset, size_t size)
+static int do_erase(const struct shell *shell, off_t offset, size_t size)
 {
 	int ret;
 
 	ret = flash_write_protection_set(flash_device, false);
 	if (ret) {
-		printk("flash_write_protection_set: %d (can't disable)\n", ret);
+		PR_ERROR(shell, "flash_write_protection_set: %d (can't"
```

Comment:
here message is misleading, better would be "Failed to disable flash protection (err: %d)."